addons/website_form/views/snippets/s_website_form.xml

```diff
@@ -133,8 +133,74 @@
                            data-apply-to=".s_website_form_date input"/>
             <we-checkbox string="Required" data-name="required_opt" data-no-preview="true"
                 data-toggle-required="s_website_form_required"/>
-            <we-checkbox string="Hidden" data-name="hidden_opt" data-no-preview="true"
-                data-select-class="s_website_form_field_hidden"/>
+            <we-select string="Visibility" data-no-preview="true">
+                <we-button data-visibility="">Always Visible</we-button>
+                <we-button data-visibility="s_website_form_field_hidden">Hidden</we-button>
+                <we-button data-visibility="s_website_form_field_hidden_if">Visible only if</we-button>
+            </we-select>
+            <we-row data-name="hidden_opt" string="">
+                <we-select data-name="hidden_condition_opt" data-no-preview="true">
+                    <!-- Load every existing form input -->
+                </we-select>
+                <we-select data-name="hidden_condition_no_text_opt" data-no-preview="true">
+                    <we-button data-visibility-comparaison="selected">Is equal to</we-button>
+                    <we-button data-visibility-comparaison="!selected">Is not equal to</we-button>
+                </we-select>
+                <we-select data-name="hidden_condition_text_opt" data-no-preview="true">
+                    <!-- str comparaison possibilities -->
+                    <we-button data-visibility-comparaison="contains">Contains</we-button>
+                    <we-button data-visibility-comparaison="!contains">Doesn't contain</we-button>
+                    <we-button data-visibility-comparaison="equal">Is equal to</we-button>
+                    <we-button data-visibility-comparaison="!equal">Is not equal to</we-button>
+                    <we-button data-visibility-comparaison="set">Is set</we-button>
+                    <we-button data-visibility-comparaison="!set">Is not set</we-button>
+                </we-select>
+                <we-select data-name="hidden_condition_num_opt" data-no-preview="true">
+                    <!-- number comparaison possibilities -->
+                    <we-button data-visibility-comparaison="equal">Is equal to</we-button>
+                    <we-button data-visibility-comparaison="!equal">Is not equal to</we-button>
+                    <we-button data-visibility-comparaison="greather">Is greather than</we-button>
```

Comment:
I don't think [this](https://www.urbandictionary.com/define.php?term=Greather) is what you mean? :smile: => Greater :wink: But maybe you could write those by their symbol (=, !=, >, <, >=, <=) ? Seems more readable and those actually shorter string so it's better.

Reply:
Some symbols are not allowed as XML attribute values and throws the error "XMLSyntaxError: Unescaped '<' not allowed in attributes values". So I think it's better to keep the current way of doing things without spelling mistakes of course :)

Reply:
or escape it like &gt; ?

Reply:
Yes I can escape the problematic symbols but then another problem arises. What do I do with comparators that have no symbol equivalent? (between, is set, ...) I think that having a mix of symbol and text comparators is not the best solution.